### PR TITLE
Intro patches

### PR DESCRIPTION
**Pull request by agitter**
I corrected errors, added TODOs from #246, and fixed the references so that this will build without error.  I'd like to merge this clean version of #246 before one of us goes through and merges the two version of the intro.

### CODE REVIEW

**Comment by agitter** [comment]
The Travis build does pass now.  There isn't any major content change here so anyone should feel free to review and I'll merge.

File: sections/02_intro.md [hunk 1]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
 One version:
-[This section needs citations]
+`TODO: this section needs citations`
 
-Deep Learning is a collection of new techniques that together have recently demonstrated 
-breakthrough gains over existing approaches in several fields. 
+Deep Learning is a collection of new techniques that together have recently
+demonstrated  breakthrough gains over existing approaches in several fields.
 
 Deep learning is built on a very old idea, neural networks, that was first
```
Comment (agapow) [inline sections/02_intro.md]: Suggest merging these two paragraphs for "sound", e.g. Deep learning is a collection ... It is built upon ...

File: sections/02_intro.md [hunk 2]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-to build and train complicated. In the past, neural networks required very specialized knoweldge to 
-build and modify, including the ability to robustly code differentials of matrix 
-expressions. Errors here are often subtle and difficult to detect, so it could be 
-very difffult to tailor networks to specific problems without substantial experience
-and training. Now, however, with these new packages, even very complex neural networks 
-are automatically differentiated, and high level scripting instructions can transparently
-run very efficently on GPUs. The technology has progressed to the point that even
-algorithms can be differentiated [cite neural stack and neural memory papers].
-
-Second, key technical insight has been uncovered that guides the construction of
```
Comment (agapow) [inline sections/02_intro.md]: I'd suggest that this second point goes first. The ability to make more complicated networks has lead to the availability of packages. Historical and logical order.

File: sections/02_intro.md [hunk 3]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-algorithms can be differentiated [cite neural stack and neural memory papers].
-
-Second, key technical insight has been uncovered that guides the construction of 
-much more complicated networks that previously possible. In the past, most neural networks 
-included just a single network. A network with an arbitrary number of hidden nodes, but 
-just a single layer, can learn arbitrarily complex functions. And networks with more than one hidden
-layer (deep networks), were hard to train. However, it turns out, deep networks can more
-efficiently represent many tasks when they are built to mirror the underlying structure of the data.
-Moroever, deep networks are more robust and trainable when employing several
-architectural innovations: weight replication, better behaived non-linearities like rectified-linear units, residual networks,
```
Comment (agapow) [inline sections/02_intro.md]: behaved

File: sections/02_intro.md [hunk 4]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-just a single layer, can learn arbitrarily complex functions. And networks with more than one hidden
-layer (deep networks), were hard to train. However, it turns out, deep networks can more
-efficiently represent many tasks when they are built to mirror the underlying structure of the data.
-Moroever, deep networks are more robust and trainable when employing several
-architectural innovations: weight replication, better behaived non-linearities like rectified-linear units, residual networks, 
-and better weight initialization, and persistent memory. Likewise the central role of 
-dimensionality reduction as a strength of neural networks was elucidated, and this has
-motivated designs built to capitalize on this strength [cite autoencoders and word2vec].
-
-Third, several advances in training algorithms have enabled applications of neural networks in ways
```
Comment (agapow) [inline sections/02_intro.md]: Actually, put the software package point last, as per previous point.

File: sections/02_intro.md [hunk 5]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-First, in scientific contexts understanding the patterns in the data may be just as important as fitting the data.
-For this reason, interpretability can be more important here than other domains. Scientific work often 
-aimes to understand the underlying principles behind the data we see, and architectures and techniques that expose the
-non-obvious patterns in the data are particualrly important and very active area of research [cite examples from all sections].
-
-Second, there are important and pressing questions about how to build networks that can efficently represent 
-the underlying logic of the data. This concern of "representability" is important, because it gives insight into
-the structure of scientific data, and when understood can guide the design of more efficent and effective networks. For example,
-one particularly important study was published in Science, which demonstrates that a simple neural network can very efficiently and
-accuratly model (i.e. respresent) a theoretically important quantum mechanical system [http://science.sciencemag.org/content/355/6325/602].
```
Comment (agapow) [inline sections/02_intro.md]: accurately

File: sections/02_intro.md [hunk 6]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-
-First, in scientific contexts understanding the patterns in the data may be just as important as fitting the data.
-For this reason, interpretability can be more important here than other domains. Scientific work often 
-aimes to understand the underlying principles behind the data we see, and architectures and techniques that expose the
-non-obvious patterns in the data are particualrly important and very active area of research [cite examples from all sections].
-
-Second, there are important and pressing questions about how to build networks that can efficently represent 
-the underlying logic of the data. This concern of "representability" is important, because it gives insight into
-the structure of scientific data, and when understood can guide the design of more efficent and effective networks. For example,
-one particularly important study was published in Science, which demonstrates that a simple neural network can very efficiently and
```
Comment (agapow) [inline sections/02_intro.md]: Maybe elide the "published in Science" bit.

File: sections/02_intro.md [hunk 7]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-understand how to judge success or failure. There is a great deal of excitement about deep learning, but in most scientific corners
-careful thought needs to be put into bringing deep learning alongside existing experts and efforts.
-
-Fourth, data availability and complexity is unevenly distributed accross science. Some areas of science like genomics and particle physics are swamped in petabytes and 
-exobytes of high quality data. Others, like chemistry, are comparatively data poor with well developed domain specific and effective algorithms. These
-differences become consequential and define the most successul approachs. For example, the convergence of lower amounts of data
-and important nuances to the domain might favor lower parameter networks that incorporate domain specific knowledge and fuse data of multiple
-different types. This flexibility, remember, is one of the most striking strengths of neural networks. In the long run,
-it is an open question the most effect strategies will be, but in this time of creative experimenation optimism is justified.
```
Comment (agapow) [inline sections/02_intro.md]: "what the most effective strategies will be" or (better) "which will be the most effective strategies"

File: sections/02_intro.md [hunk 8]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
-
-None of these scientific concerns should dampen enthusiasm about deep learning. Rather, because the approaches flexibility,
```
Comment (agapow) [inline sections/02_intro.md]: Are they scientific or methodological concerns?

File: sections/02_intro.md [hunk 9]
```diff
@@ -34,88 +34,118 @@ learning methods more challenging or less fruitful to apply.
...
+thought stretching back decades and even centuries. Deep learning will always be
+in dialogue with this expertise, to understand the key problems, encode the most
+salient prior knowledge, and understand how to judge success or failure. There
+is a great deal of excitement about deep learning, but in most scientific
+corners careful thought needs to be put into bringing deep learning alongside
+existing experts and efforts. `TODO: appropriate comparisons with state of the
+art`
+
+Fourth, data availability and complexity is unevenly distributed across science.
+Some areas of science like genomics and particle physics are swamped in
```
Comment (agapow) [inline sections/02_intro.md]: Astronomy is the other bug data generator. http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002195 is a nice paper on big data in biology, c.v. "4-headed beast"

File: sections/02_intro.md (hunk 1 above)
Comment (agitter) [inline sections/02_intro.md]: Done

File: sections/02_intro.md (hunk 2 above)
Comment (agitter) [inline sections/02_intro.md]: Saving until after merge

File: sections/02_intro.md (hunk 3 above)
Comment (agitter) [inline sections/02_intro.md]: Already fixed

File: sections/02_intro.md (hunk 4 above)
Comment (agitter) [inline sections/02_intro.md]: Saving until after merge

File: sections/02_intro.md (hunk 5 above)
Comment (agitter) [inline sections/02_intro.md]: Already fixed

File: sections/02_intro.md (hunk 6 above)
Comment (agitter) [inline sections/02_intro.md]: Already fixed

File: sections/02_intro.md (hunk 7 above)
Comment (agitter) [inline sections/02_intro.md]: Changed

File: sections/02_intro.md (hunk 8 above)
Comment (agitter) [inline sections/02_intro.md]: Cut scientific

File: sections/02_intro.md (hunk 9 above)
Comment (agitter) [inline sections/02_intro.md]: Saving until after we address the TODO about the scale of the data not being the only factor

**Comment by agitter** [comment]
Thanks @agapow.  I fixed the typos and grammatical problems.  The content changes I'm saving until we merge the two versions of the intro.  Some of this text may be cut.

**Comment by cgreene** [comment]
Looks good to me given that concerns will be addressed after merge. 👍

**Comment by agitter** [comment]
I'm going to merge this.  The changes were minor, and I want to have a working Travis build before I make changes to other sections.  There will be an opportunity to comment on the introduction again when we create a pull request to merge the two versions.

**Pull request merged**